src/App.js

```diff
@@ -41,6 +50,18 @@ class App extends React.Component {
     // TODO: update ONE todo in state using the id
     // make sure not to mutate the existing array!
     // also, make sure not to mutate any objects within the array!
+    const updatedTodos = this.state.todos.map(todo => {
+      if (todo.id === id){
+        const updatedTodo = { ...todo }
+        updatedTodo.completed = completed 
+        return updatedTodo
```

Comment:
Very nice! If you want a slightly shorter syntax, you can also do this: 

```js
// spread the old todo into a new object
// overwrite the key of "completed" with the value of completed from updateTodo function's argument
return { ...todo, completed } 
```

But the way you have it written works the same.